docs/ru/amls.md

```diff
@@ -0,0 +1,711 @@
+	if (gps_rotating_mem_location == 35)
+		// Start at 0 when the memory location 35 is reached
+		gps_rotating_mem_location = 0;
+
+	// Remember the error for the next loop
+	gps_lat_error_previous = gps_lat_error;
+	gps_lon_error_previous = gps_lon_error;
+
+	//Calculate the GPS pitch and roll correction as if the nose of the multicopter is facing north.
+	//The Proportional part = (float)gps_lat_error * gps_p_gain.
+	//The Derivative part = (float)gps_lat_total_avarage * gps_d_gain.
+	gps_pitch_adjust_north = (float)gps_lat_error * gps_p_gain + (float)gps_lat_total_avarage * gps_d_gain;
+	gps_roll_adjust_north = (float)gps_lon_error * gps_p_gain + (float)gps_lon_total_avarage * gps_d_gain;
+
+	if (!latitude_north)
+		// Invert the pitch adjustmet because the quadcopter is flying south of the equator
+		gps_pitch_adjust_north *= -1;
+	if (!longiude_east)
+		// Invert the roll adjustmet because the quadcopter is flying west of the prime meridian
+		gps_roll_adjust_north *= -1;
+
+	//Because the correction is calculated as if the nose was facing north, we need to convert it for the current heading.
+	gps_roll_adjust = ((float)gps_roll_adjust_north * cos(angle_yaw * 0.017453)) + ((float)gps_pitch_adjust_north * cos((angle_yaw - 90) * 0.017453));
+	gps_pitch_adjust = ((float)gps_pitch_adjust_north * cos(angle_yaw * 0.017453)) + ((float)gps_roll_adjust_north * cos((angle_yaw + 90) * 0.017453));
+
+	//Limit the maximum correction to 300. This way we still have full controll with the pitch and roll stick on the transmitter.
+	if (gps_roll_adjust > 300) gps_roll_adjust = 300;
+	if (gps_roll_adjust < -300) gps_roll_adjust = -300;
+	if (gps_pitch_adjust > 300) gps_pitch_adjust = 300;
+	if (gps_pitch_adjust < -300) gps_pitch_adjust = -300;
+}
+
+```
+
+<a name="2-gps-following"></a>
+
+## 2. Следование за платформой
+
+Основной задачей стабилизации по GPS-координатам стала разработка алгоритма предсказания положения дрона. Самым простым способом представилось использовать математический расчет следующего положения дрона. Это вычисляется для наиболее точного позиционирования дрона в отношении посадочной платформы.
+
+Для начала был придуман простейший алгоритм расчета коэффициента изменения координат. Реализация производилась на языке Python. На этапе тестирования данного алгоритма встала проблема симуляции генерации GPS-координат. Дабы разрешить эту проблему, было испробовано много различных ресурсов: от открытых исходных кодов самодельных навигаторов до попытки использовать API Google Maps, Yandex Maps или 2GIS. И лишь спустя семестр мы додумались до простого изменения значений по некоторой дельте с отрисовкой в MatPlotLib либо PyQtGraph. До этого всё тестирование алгоритма производилось с использованием инструментария прошивки PX4, симулятора движения дрона Gazebo. Как следствие было преодолено много формальностей в вопросах общения с симулятором и увеличением производительности.
+
+Видео работы алгоритма предсказания GPS координат (кликабельно)
```

Comment:
Wouldn't it be better to embed the videos into the article using iframe? It's done like here:

```html
<iframe width="560" height="315" src="https://www.youtube.com/embed/HdHbZFz7nR0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

You also won't need preview images in the repo.

Reply:
Sorry, but the code with the iframe is displayed as html code and not as video. It didn't work.

Reply:
It won't work on GitHub, but it will on the main site clover.coex.tech. The site is the main way of viewing this documentation.